### ISSUE TITLE
Back button

### ISSUE THREAD

**Issue by apod42** [mrk-j/paginga#8]
Hello and thank you for a really good script! One issue though... Let's say I navigate to page 3, and here I press a link that takes me to an external page. If I press the back button in the browser, I am taken to page 1, not page 3. 

This is browser dependent. Safari on Ios and Firefox on windows take me to page 3. Chrome and Edge take me to page 1. 

Is there a fix for this? An option would be to somehow add a hash to the URL. Could there be a more elegant solution?

**Comment by mrk-j** [comment]
Hey @apod42, thanks for using paginga and taking the time to let me know were it can be improved! I will look into this in the coming days. I think the hash is the most suitable solution, this way we also create an option to create unique URL's to specific pages. The only problem I currently see is that when there are multiple instances of paginga on the same page.

**Comment by apod42** [comment]
Hello and thanks for the answer! I also tryed the hash approch but I'm doing something wrong. If for example I do this:

previousElement.on("click", function()
                {
                    plugin.showPreviousPage.call(plugin);
                    window.location.hash = '#1';

```
            });

            nextElement.on("click", function()
            {
                plugin.showNextPage.call(plugin);
                window.location.hash = '#2';

            });

            firstElement.on("click", function()
            {
                plugin.showFirstPage.call(plugin);
                window.location.hash = '#3';

            });

            lastElement.on("click", function()
            {
                plugin.showLastPage.call(plugin);
                window.location.hash = '#4';

            });
```

The hash is added to the URL but when I press the back button or navigate with back/forward, even though the url cahges I always get page 1 displayed. What am I doing wrong?

**Comment by mrk-j** [comment]
@apod42 Can you checkout my fix in #9? If you set the `history` option to `true` it will work as expected as far as I can see.

**Comment by apod42** [comment]
Ok did some testing. Works as intended for page numbers (when I navigate by pressing 1 or 2 or 4 etc). 

Navigation by arrows (next, previous, first, last) does not add hash to URL. If I add plugin.setHistoryUrl.call(plugin); to lastElement.on("click", function(), firstElement.on("click", function(), and so on, hash numbers in the URL do not correspond to the current page I am on. They are random at least.

**Comment by mrk-j** [comment]
It needed to be added to the `show*Page` functions: https://github.com/mrk-j/paginga/pull/9/commits/917ac5c9e74ca1bfed9c9d1640d397b101e00669

It should be fixed now if you get the latest code from #9.

**Comment by apod42** [comment]
Look really good. Thank you for the feature addon, I think you can close this now.

Will to further testing in the following days when I have my project ready. Will get back to you if I discover some bugs, but for now works great!

**Comment by mrk-j** [comment]
Ok, I will close this issue. I will merge the PR when I have tested it with multiple instances on one page. Also I think I'm going to create the possibility to add a prefix to the page number.

**Comment by mrk-j** [comment]
I just merged #9 and tagged `v0.8`!
